org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/RenameHandler.java

```diff
@@ -81,8 +84,12 @@ public WorkspaceEdit rename(RenameParams params, IProgressMonitor monitor) {
 			}
 			RenameRefactoring renameRefactoring = renameSupport.getRenameRefactoring();
 
-			CreateChangeOperation create = new CreateChangeOperation(new CheckConditionsOperation(renameRefactoring, CheckConditionsOperation.ALL_CONDITIONS), RefactoringStatus.FATAL);
+			CheckConditionsOperation check = new CheckConditionsOperation(renameRefactoring, CheckConditionsOperation.ALL_CONDITIONS);
+			CreateChangeOperation create = new CreateChangeOperation(check, RefactoringStatus.FATAL);
 			create.run(monitor);
+			if (check.getStatus().getSeverity() >= RefactoringStatus.ERROR) {

```

Comment:
if (check.getStatus().hasFatalError()) {

otherwise, a rename refactoring operation will fail if there is an invalid rename participant.

That's why the following four tests fail in Eclipse - testRenamePackage, testRenameType, testRenameOverrideMethodSimple and testRenameField.

The testRenameTypeWithResourceChanges test still fails in Eclipse, if the workspace name contains "E".

Could you try to run RenameHandlerTest in Eclipse using the workspace '${workspace_loc}/../junit-workspacE' name ?